### PR TITLE
`BlockchainInterface` cleanups

### PR DESCRIPTION
**Pull request by arvidn**
This PR is best reviewed one commit at a time.

### Purpose:

Minor simplifications of protocols.

### Current Behavior:

* The `BlockInfo` protocol has implementations for its functions
* `find_fork_point_in_chain()` and `lookup_fork_chain()` takes `BlockchainInterface`
* `WalletBlockchain` implements `BlockchainInterface`

### New Behavior:

* The `BlockInfo` protocol has empty functions
* `find_fork_point_in_chain()` and `lookup_fork_chain()` takes the smaller `FullBlocksProtocol`
* `WalletBlockchain` implements `BlockRecordsProtocol`

### Testing Notes:

### CODE REVIEW

File: chia/_tests/wallet/sync/test_wallet_sync.py [hunk 1]
```diff
@@ -1450,9 +1450,6 @@ async def test_bad_peak_mismatch(
         await asyncio.sleep(3)
         await wallet_server.start_client(PeerInfo(self_hostname, full_node_server.get_port()), None)
         await wallet_node.new_peak_wallet(msg, wallet_server.all_connections.popitem()[1])
-        await asyncio.sleep(3)
-        assert wallet_node.wallet_state_manager.blockchain.get_peak_height() != fake_peak_height
```
Comment (almogdepaz) [inline chia/_tests/wallet/sync/test_wallet_sync.py:1454]: this looks important for the test

Comment (arvidn) [reply chia/_tests/wallet/sync/test_wallet_sync.py:1454]: `get_peak_height()` returns `None` unconditionally on `WalletBlockchain`. I don't think this test does anything.
I suppose I could change it to ensure `fake_peak_heights is not None`.

https://github.com/Chia-Network/chia-blockchain/pull/18522/files#diff-7a24ec350c9cfa1f41bb4b59112b017f1523594d4cf7d42539880807c95dbcefL218-L219

Comment (arvidn) [reply chia/_tests/wallet/sync/test_wallet_sync.py:1454]: oh, `fake_peak_height` is assigned a value just a few lines above. that seems meaningless to test

**Review by almogdepaz — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok but would be nice to fix the test

**Pull request merged**